File: scala/165.scala
object Solution {
    def compareVersion(version1: String, version2: String): Int = {
        val v1 = version1.split("\\.")
        val v2 = version2.split("\\.")
        def cmp(versions1: Array[String], versions2: Array[String]): Int = {
            if (versions1.isEmpty && versions2.isEmpty) 0
            else if (versions1.isEmpty) {
                cmp(versions1 :+ "0", versions2)
            } else if (versions2.isEmpty) {
                cmp(versions1, versions2 :+ "0")
            }
            else {
                val a = versions1.head.toInt
                val b = versions2.head.toInt
                if (a > b) 1
                else if (a < b) -1
                else cmp(versions1.drop(1), versions2.drop(1))
            }
        }
        cmp(v1, v2)
    }
}

File: leetcodeJava/165.java
package leetcodeJava;

class Solution {
    public int compareVersion(String version1, String version2) {
        var v1 = version1.split("\\.");
        var v2 = version2.split("\\.");
        int i = 0, j = 0;
        while (i < v1.length || j < v2.length) {
            var n = 0;
            if (i < v1.length)
                n = Integer.valueOf(v1[i]).intValue();
            var m = 0;
            if (j < v2.length)
                m = Integer.valueOf(v2[j]).intValue();
            if (n > m) return 1;
            else if (n < m) return -1;
            i++;
            j++;
        }
        return 0;
    }
}
